JeMPI_Apps/JeMPI_UI/src/hooks/useConfig.tsx

```diff
@@ -0,0 +1,67 @@
+import { useQuery } from '@tanstack/react-query'
+import { AxiosError } from 'axios'
+import React, { useMemo } from 'react'
+import Loading from '../components/common/Loading'
+import ApiErrorMessage from '../components/error/ApiErrorMessage'
+import { ApiClient } from '../services/ApiClient'
+import getConfig, { Config } from 'config'
+import { getApiClient } from '../services/ApiClient'
+
+export interface ConfigContextValue {
+  config: Config
+  apiClient: ApiClient
+}
+
+const ConfigContext = React.createContext<ConfigContextValue | null>(null)
+ConfigContext.displayName = 'ConfigContext'
+
+export interface ConfigProviderProps {
+  children: React.ReactNode
+}
+
+export const ConfigProvider = ({
+  children
+}: ConfigProviderProps): JSX.Element => {
+  const {
+    data: config,
+    error: configError,
+    isLoading: isLoadingConfig,
+    isError: isConfigError
+  } = useQuery<Config, AxiosError>({
+    queryKey: ['config'],
+    queryFn: async () => getConfig(),
+    refetchOnWindowFocus: false
+  })
+
+  const apiClient = useMemo(
+    () => config && getApiClient(config),
+    [config]
+  ) as unknown as ApiClient
+
+  if (isLoadingConfig || !config) {
+    return <Loading />
+  }
+
+  if (isConfigError) {
```

Comment:
Should we not also error on `!isLoadingConfig && !apiClient` in case something went wrong with the getting the api Client

Reply:
We handle the API error ` if (isConfigError)`.